randovania/gui/multiplayer_session_window.py

```diff
@@ -750,13 +753,19 @@ def generate_layout(progress_update: ProgressUpdateCallable):
             except AnyNetworkError:
                 # We're interested in catching generation failures.
                 # Let network errors be handled by who called us, which will be captured by handle_network_errors
+                alert_user(self, self._options)
+                # Alert the person who generates on errors since update_game_tab
+                # doesn't generation errors
                 raise
 
             except Exception as e:
                 await self.failure_handler.handle_exception(
                     e,
                     self.update_progress,
                 )
+                alert_user(self, self._options)
```

Comment:
If the intent is to notify when it finished, error or not, you can use the `finally` block

Reply:
We don't want to alert on cancel error or on success

Reply:
On cancel, yeah.
On success, not sure? You can definitely leave it generating and alt-tab while it runs. Especially for big multis.

Reply:
the host will still be notified on success, just not via this function. It's from [`update_game_tab`](https://github.com/randovania/randovania/pull/7465/files#diff-53f4482d0e5d76127403793fa68c960d5c93cd78fd4646fb7fea40d2989241aaR452) instead.

Reply:
Yeah, on testing, doing it here, it only alerts the host. 
Doing it with update_game_tab alerts everyone.